code/modules/power/supermatter/supermatter.dm

```diff
@@ -450,6 +473,37 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		l.hallucination += power * config_hallucination_power * D
 		l.hallucination = CLAMP(0, 200, l.hallucination)
 
+	// Switch the distortion effect based on the crystal's state
+	if(damage != damage_archived)
+		cut_overlay(distort)
+		switch(get_status())
+			if(SUPERMATTER_INACTIVE)
+				distort.icon = 'icons/effects/96x96.dmi'
+				distort.icon_state = "SM_base"
+				distort.pixel_x = -32
+				distort.pixel_y = -32
+			if(SUPERMATTER_NORMAL, SUPERMATTER_NOTIFY, SUPERMATTER_WARNING)
+				distort.icon = 'icons/effects/96x96.dmi'
+				distort.icon_state = "SM_base_active"
+				distort.pixel_x = -32
+				distort.pixel_y = -32
+			if(SUPERMATTER_DANGER)
+				distort.icon = 'icons/effects/160x160.dmi'
+				distort.icon_state = "SM_delam_1"
+				distort.pixel_x = -64
+				distort.pixel_y = -64
+			if(SUPERMATTER_EMERGENCY)
+				distort.icon = 'icons/effects/224x224.dmi'
+				distort.icon_state = "SM_delam_2"
+				distort.pixel_x = -96
+				distort.pixel_y = -96
+			if(SUPERMATTER_DELAMINATING)
+				distort.icon = 'icons/effects/288x288.dmi'
+				distort.icon_state = "SM_delam_3"
+				distort.pixel_x = -128
+				distort.pixel_y = -128
+		add_overlay(distort)
```

Comment:
This needs to be done inside of update_overlays, otherwise you may lose the overlay when that is called for unrelated reasons.
Add this code to update overlays, then call update_icon(UPDATE_OVERLAYS) (Might not be the exact wording for the proc) to prevent this.

Reply:
Done